tests/scripts/all-core.sh

```diff
@@ -131,22 +132,8 @@ pre_set_shell_options () {
     set -e -o pipefail -u
 }
 
-# For project detection
-in_mbedtls_repo () {
-    test "$PROJECT_NAME" = "Mbed TLS"
-}
-
-in_tf_psa_crypto_repo () {
-    test "$PROJECT_NAME" = "TF-PSA-Crypto"
-}
-
 pre_check_environment () {
-    # For project detection
-    PROJECT_NAME_FILE='./scripts/project_name.txt'
-    if read -r PROJECT_NAME < "$PROJECT_NAME_FILE"; then :; else
-        echo "$PROJECT_NAME_FILE does not exist... Exiting..." >&2
-        exit 1
-    fi
+    . framework/scripts/project_detection.sh
```

Comment:
I think we need something a bit more sophisticated here:
- when we're in tf-psa-crypto as a submodule/subdirectory of mbedtls (which is the only way now), we want to grab the framework from mbedtls (that's both the only option now, and apparently the right thing to do in the long run too);
- when we're in tf-psa-crypto on its own (which is not possible now but will become an option before then end of the year) we want to use the framework from there (it will be added as a submodule when tf-psa-crypto becomes standalone).

So I'd suggest something like
```sh
if [ -d ../include/mbedtls && -r ../framework/scripts/project_detection.sh ]; then
    # we're in tf-psa-crypto as a submodule of mbedtls, grab the framework from mbedtls
    . ../framework/scripts/project_detection.sh
else
    # we're in tf-psa-crypto standalone or in mbedtls, use the local framework
    . framework/scripts/project_detection.sh
fi
```

Cc @ronald-cron-arm

Reply:
The intent is correct to me. For the way we do it, we could have a variable defined in the mbedtls/tf-psa-crypto all.sh that defines the path from root to the framework and use it here?